composite-actions/nodejs-script/service-definition-validator/src/utils/markdownGenerator.ts

```diff
@@ -0,0 +1,100 @@
+import type { ValidationResult } from "../types/index.ts";
+import { REPORT_COMMENT_IDENTIFIER } from "./constants.ts";
+
+/**
+ * Capitalizes the first letter of a string.
+ * @param text - The string to capitalize.
+ * @returns The capitalized string or the original string if empty.
+ */
+function capitalize(text: string): string {
+  return text.charAt(0).toUpperCase() + text.slice(1);
+}
+
+/**
+ * Generates a Markdown report based on validation results.
+ * @param results - Array of ValidationResult objects.
+ * @returns Markdown formatted string.
+ */
+export function generateMarkdownReport(results: ValidationResult[]): string {
+  const hasIssues = results.some(
+    (result) => result.missingVars.length > 0 || result.mismatchedVars.length > 0,
+  );
+
+  const mainIcon = hasIssues ? "⚠️" : "✅";
```

Comment:
Not to make more hassle with this, but we defined 3 levels of having issues:

* none == green check mark
* warning == yellow exclamation mark
* error == red x or red exclamation mark

Should we have the same here? And show the most severe in the heading? A warning is allowed, but if the warning "hides" an error, then perhaps it is good to be explicit in the "summary" too?